src/projectM-sdl/projectM_SDL_main.cpp

```diff
@@ -315,53 +342,58 @@ srand((int)(time(NULL)));
     projectMSDL *app;
     
     std::string base_path = DATADIR_PATH;
-    SDL_LogInfo(SDL_LOG_CATEGORY_APPLICATION, "Using data directory: %s\n", base_path.c_str());
 
-    // load configuration file
+    // load configuration file folder (it might be different than DATADIR_PATH)
     std::string configFilePath = getConfigFilePath(base_path);
 
-    if (! configFilePath.empty()) {
-        // found config file, use it
-        app = new projectMSDL(configFilePath, 0);
-        SDL_Log("Using config from %s", configFilePath.c_str());
-    } else {
-        base_path = SDL_GetBasePath();
-        SDL_Log("Config file not found, using built-in settings. Data directory=%s\n", base_path.c_str());
-
-		// Get max refresh rate from attached displays to use as built-in max FPS.
-		int i = 0;
-		int maxRefreshRate = 0;
-		SDL_DisplayMode current;
-		for (i = 0; i < SDL_GetNumVideoDisplays(); ++i)
+    if (configFilePath.empty()) base_path = SDL_GetBasePath(); // if no configfile was found, use SDL base path.
+	if (configFilePath != base_path) base_path = configFilePath; // if config file was found in a different directory than DATADIR_PATH, change base path to it.
+
+	SDL_LogInfo(SDL_LOG_CATEGORY_APPLICATION, "Using data directory: %s\n", base_path.c_str());
+
+	// Get max refresh rate from attached displays to use as built-in max FPS.
+	int i = 0;
+	int maxRefreshRate = 0;
+	SDL_DisplayMode current;
+	for (i = 0; i < SDL_GetNumVideoDisplays(); ++i)
+	{
+		if (SDL_GetCurrentDisplayMode(i, &current) == 0)
 		{
-			if (SDL_GetCurrentDisplayMode(i, &current) == 0)
-			{
-				if (current.refresh_rate > maxRefreshRate) maxRefreshRate = current.refresh_rate;
-			}
+			if (current.refresh_rate > maxRefreshRate) maxRefreshRate = current.refresh_rate;
 		}
-		if (maxRefreshRate <= 60) maxRefreshRate = 60;
-
-        float heightWidthRatio = (float)height / (float)width;
-        projectM::Settings settings;
-        settings.windowWidth = width;
-        settings.windowHeight = height;
-        settings.meshX = 128;
-        settings.meshY = settings.meshX * heightWidthRatio;
-		settings.fps = maxRefreshRate;
-        settings.smoothPresetDuration = 3; // seconds
-        settings.presetDuration = 22; // seconds
-        settings.beatSensitivity = 0.8;
-        settings.aspectCorrection = 1;
-        settings.shuffleEnabled = 1;
-        settings.softCutRatingsEnabled = 1; // ???
-        // get path to our app, use CWD or resource dir for presets/fonts/etc
-        settings.presetURL = base_path + "presets";
+	}
+	if (maxRefreshRate <= 60) maxRefreshRate = 60;
+
+    float heightWidthRatio = (float)height / (float)width;
+    projectM::Settings settings;
+    settings.windowWidth = width;
+    settings.windowHeight = height;
+    settings.meshX = 128;
```

Comment:
can we define these defaults in one place? seemed like they were specified above too

Reply:
> can we define these defaults in one place? seemed like they were specified above too

It will only be defined once in SDL. Those settings will be used as the built-in fall back ORalso sent with your config file. That way if your config file is missing anything (most likely on purpose because you commented out things like Width / Height) then it will fall back on the built-in SDL settings.

Defaults are also defined in libProjectM and I kept those the same.